extensions/functions_arithmetic.yaml

```diff
@@ -922,6 +922,69 @@ aggregate_functions:
         decomposable: MANY
         intermediate: fp64?
         return: fp64?
+  - name: "sum0"
+    description: Sum a set of values. The sum of zero elements yields zero.
```

Comment:
An alternative is to have one sum function with an option for how to handle the set of zero inputs.  Potential complications of course arise if we want to add an option after the fact but is that something we are interested in?

Reply:
I see your point. Let’s discuss further. 
Options is interesting and that could simplify things, but not sure how to represent nullability in return statements. 

cc @westonpace @jacques-n

Reply:
It's also pretty easy to get the behavior already with `COALESCE(SUM(x), 0)` (though this is not exactly the same because it would return 0 for the sum of all null input where this function would hopefully return null).  However, I don't think we need to shy away from functions that can be represented elsewhere.  So I think adding a new function or option is still a good idea.

I'm not sure where exactly to draw the line between "new function" and "option to an existing function".  I don't have a really strong opinion either way.

Reply:
Actually I think it would behave the same as `COALESCE(SUM(x), 0)` . It is my understanding that if you had a set of all null values and you do a SUM0() on them, you would get 0, not null.

This is a description I found for SUM0 that I found here https://www.tabnine.com/code/java/classes/org.apache.calcite.sql.fun.SqlSumEmptyIsZeroAggFunction
```
Sum0 is an aggregator which returns the sum of the values which go into it like Sum. It differs in that when no non null values are applied zero is returned instead of null. 
```

Reply:
Then let's update the description to something like...

```
Sum a set of values. The sum of zero elements yields zero.  Null values are ignored.
```

We should probably clarify this for `sum` as well but that can be left for a future PR.

Reply:
@westonpace And I think we should probably leave the return type without nulls. Is that okay?

Reply:
Yes, good catch.  Given this updated understanding I think the return type should be `DECALRED_OUTPUT` nullability and a non-nullable return type.

Reply:
Got it. I will update the PR. Thanks @wmalpica for your feedback as well.